internal/kafka/internal/services/data_plane_kafka.go

```diff
@@ -379,6 +427,35 @@ func (d *dataPlaneKafkaService) persistKafkaRoutes(kafka *dbapi.KafkaRequest, ka
 	return nil
 }
 
+func getStatus(status *dbapi.DataPlaneKafkaStatus) managedKafkaStatus {
```

Comment:
Side comment about this:

I see `getStatus` and other functions in this file like `getRoutes` are at package-level. I don't see any reason why this should be that way. The smaller the scope the better. Can we make them part of the `dataplaneKafkaService` type? We can deal with that in another PR if you prefer

Reply:
No reason they couldn't be part of the dataplane kafka service. Let me look at the possibility of that.
This comment also made me realise that it should `getManagedKafkaStatus` instead of `getStatus`

Reply:
Done in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/1320/commits/d58b78c0b1546b973612df26366c9cc4e6313728

Thanks for the suggestion @miguelsorianod